### PR TITLE
Sign Up Post Request to Backend

### PR DESCRIPTION
**Pull request by thywang**
## Notion ticket link

[Sign Up Post Request to Backend](https://www.notion.so/uwblueprintexecs/Sign-Up-Post-Request-to-Backend-154810dfa8d347bdbc7bc60a738959ff?pvs=4)

## Implementation description
* Add phone number and emergency phone number to user model
* Link the sign up components as a multistep form
* Update functions to reflect changes in user model
* Make a post request with the sign up info

## Steps to test
1. Navigate to /signup
2. Follow the steps to sign up
3. Verify in MongoDB test.users the new user is created

## What should reviewers focus on?
* Design adheres with Figma design
* State is preserved when going back in sign up process (note: will require confirming email each time going back)
* Appropriate alert shows when user is successfully created in MongoDB or if there is an error

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by sebmendoza — COMMENTED** [review]
I'm getting a lot of errors related to the mongoose model. It shut down my backend. Are you getting this?

They were a bunch of type errors:
`scv2_ts_backend  |   Object literal may only specify known properties, and 'phoneNumber' does not exist in type '(Without<userCreateInput, userUncheckedCreateInput> & userUncheckedCreateInput) | (Without<...> & userCreateInput)'.
scv2_ts_backend  | services/implementations/userService.ts(399,41): error TS2339: Property 'phoneNumber' does not exist on type '{ id: string; authId: string; email: string; firstName: string; lastName: string; role: Role; isAccepted: Status; }'.
scv2_ts_backend  | services/implementations/userService.ts(400,48): error TS2339: Property 'emergencyFirstName' does not exist on type '{ id: string; authId: string; email: string; firstName: string; lastName: string; role: Role; isAccepted: Status; }'.
scv2_ts_backend  | services/implementations/userService.ts(401,47): error TS2339: Property 'emergencyLastName' does not exist on type '{ id: string; authId: string; email: string; firstName: string; lastName: string; role: Role; isAccepted: Status; }'.
scv2_ts_backend  | services/implementations/userService.ts(402,50): error TS2339: Property 'emergencyPhoneNumber' does not exist on type '{ id: string; authId: string; email: string; firstName: string; lastName: string; role: Role; isAccepted: Status; }'.
scv2_ts_backend  | 
scv2_ts_backend  |     at createTSError (/app/node_modules/ts-node/src/index.ts:587:12)
scv2_ts_backend  |     at reportTSError (/app/node_modules/ts-node/src/index.ts:591:19)
scv2_ts_backend  |     at getOutput (/app/node_modules/ts-node/src/index.ts:921:36)
scv2_ts_backend  |     at Object.compile (/app/node_modules/ts-node/src/index.ts:1189:32)
scv2_ts_backend  |     at Module.m._compile (/app/node_modules/ts-node/src/index.ts:1295:42)
scv2_ts_backend  |     at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
scv2_ts_backend  |     at Object.require.extensions.<computed> [as .ts] (/app/node_modules/ts-node/src/index.ts:1298:12)
scv2_ts_backend  |     at Module.load (node:internal/modules/cjs/loader:981:32)
scv2_ts_backend  |     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
scv2_ts_backend  |     at Module.require (node:internal/modules/cjs/loader:1005:19)
scv2_ts_backend  |     at require (node:internal/modules/cjs/helpers:102:18)
scv2_ts_backend  |     at Object.<anonymous> (/app/middlewares/auth.ts:4:1)
scv2_ts_backend  |     at Module._compile (node:internal/modules/cjs/loader:1101:14)
scv2_ts_backend  |     at Module.m._compile (/app/node_modules/ts-node/src/index.ts:1295:23)
scv2_ts_backend  |     at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
scv2_ts_backend  |     at Object.require.extensions.<computed> [as .ts] (/app/node_modules/ts-node/src/index.ts:1298:12)
scv2_ts_backend  | [nodemon] app crashed - waiting for file changes before starting...`

**Comment by thywang1** [comment]
Hi @sebmendoza, thanks for reviewing! No... I tried recloning and checking out to this branch, and didn't get any errors. We could schedule a quick call to compare. It may have to do with npx prisma or Docker.

Screen Shot 2024-03-07 at 10 50 06 PM

File: frontend/src/components/auth/Signup.tsx [hunk 1]
```diff
@@ -1,45 +1,75 @@
...
+  phoneNumberError: "",
+};
 
 const Signup = (): React.ReactElement => {
-  const [firstName, setFirstName] = useState('');
-  const [lastName, setLastName] = useState('');
+  const allErrors = [SignupMainErrors, SignupSecondaryErrors, SignupEmergencyContactErrors];
+  const steps: SignupFormStepComponentType[] = [SignupMain, SignupSecondary, SignupEmergencyContact];
+
+  const { StepComponent, currentStepIndex, isFirstStep, isLastStep, back, next } = useMultistepForm(steps);
```
Comment (sebmendoza) [inline frontend/src/components/auth/Signup.tsx]: unused

**Review by sebmendoza — APPROVED** [review]
Looks good to me. Works properly. Clean solution too!!!! 

Before you merge, I have a couple questions.
I think isFirstStep is not used anywhere, why do you need it? Also, my VScode was throwing typing warning like `Parameter 'prev' implicitly has an 'any' type.ts(7006)`. If you can, just add some more types to functions.

**Review by kevinouyang1 — APPROVED** [review]
already reviewed :)

**Pull request merged**